### PR TITLE
UID2-2830 add site new app names

### PR DESCRIPTION
**Pull request by caroline-ttd**
uid2-shared: https://github.com/IABTechLab/uid2-shared/pull/225
uid2-admin: https://github.com/IABTechLab/uid2-admin/pull/264

```
{
  "id": 1004,
  "name": "kjhkjhjkm,n,n",
  "description": "",
  "enabled": false,
  "clientTypes": [],
  "domain_names": [
    "mbmnbmnb.com"
  ],
  "app_names": [
    "hgjkhljkjkbh"
  ],
  "visible": true,
  "created": 1713224480,
  "appnNames": [
    "hgjkhljkjkbh"
  ]
}
```

### CODE REVIEW

File: src/main/resources/localstack/s3/core/sites/sites.json
Comment (mcollins-ttd) [inline src/main/resources/localstack/s3/core/sites/sites.json:55]: Can we move this to site 124, the publisher?

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (mcollins-ttd) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:709]: Would be good to have a test that shows roles other than MAINTAINER and SHARING_PORTAL can't call these APIs.

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java
Comment (asloobq) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:446]: you can probably combine the stream operation with the for loop
```
List<String> normalizedAppNames = appNamesJa.stream()
                .map(String::valueOf)
                .map(String::toLowerCase)
                .collect(Collectors.toList());
```

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java
Comment (asloobq) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:370]: since `setAppNames()` accepts a Set perhaps we can return a Set from `getNormalizedAppNames()` as well ?

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java
Comment (asloobq) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:372]: I see these 3 lines repeated in `handleSiteDomains()` and `handleSiteTypesSet()`, better to keep code DRY.
```
            final List<Site> sites = this.siteProvider.getAllSites()
                    .stream().sorted(Comparator.comparingInt(Site::getId))
                    .collect(Collectors.toList());

            storeWriter.upload(sites, null);

            rc.response().end(jsonWriter.writeValueAsString(existingSite));
```

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (asloobq) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:713]: I don't think its necessary to test for all error messages that are returned by `getSiteFromParam()` since you are just testing that `handleSiteAppNames()` is called correctly. Still, no harm in adding these tests but not providing a lot of value IMO

**Review by asloobq — COMMENTED** [review]
a few small comments. otherwise LGTM

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (asloobq) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:106]: suggestion : change the app_names to be more descriptive for readability eg. 'com.android.app'

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (caroline-ttd) [reply src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:708]: I added a new test `appNameRoleUnauthorized` with role `MAPPER`

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java
Comment (caroline-ttd) [reply src/main/java/com/uid2/admin/vertx/service/SiteService.java]: That's is a good idea. We no longer need `toLowerCase` method but I will do it next time.

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java
Comment (caroline-ttd) [reply src/main/java/com/uid2/admin/vertx/service/SiteService.java]: Good catch, I created a new private class `uploadSiteToStoreWriterAndWriteExistingSiteToResponse`

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (caroline-ttd) [reply src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:712]: I just feel it is easy to write and no harm to have it LOL

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (caroline-ttd) [reply src/test/java/com/uid2/admin/vertx/SiteServiceTest.java]: That makes sense. I created `"com.123.game.app.android", "12345678"`

File: src/main/resources/localstack/s3/core/sites/sites.json
Comment (sunnywu) [inline src/main/resources/localstack/s3/core/sites/sites.json:24]: for future testing i suggest to add some case sensitiveness into these values, e.g. 

com.123.Game.App.android
com.123.Game.App.ios

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java
Comment (sunnywu) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:820]: for this test, please change the app names used here to be a mix of lower/upper cases to add extra test coverage on the case sensitivity for app names

Comment (caroline-ttd) [reply src/test/java/com/uid2/admin/vertx/SiteServiceTest.java]: I changed the app names.

File: src/main/resources/localstack/s3/core/sites/sites.json
Comment (caroline-ttd) [reply src/main/resources/localstack/s3/core/sites/sites.json]: I adjusted the value to include a mixture of lower and upper cases.

**Review by sunnywu — APPROVED** [review]

**Comment by sunnywu** [comment]
I clicked Approved but please make a new uid2 shared lib release and use it in this uid2-admin PR and ask for someone else to approve (e.g. @asloobq ) before merging

**Review by asloobq — APPROVED** [review]

**Review by asloobq — APPROVED** [review]

**Pull request merged**
